### ISSUE TITLE
Out-of-core stochastic gradient boosting for scikit-learn wrapper?

### ISSUE THREAD

**Issue by OneRaynyDay** [dmlc/xgboost#2585]
I see that there is an interface in xgb's standard api for out-of-core stochastic gradient boosting by calling this many times with different subsamples/batches of data:

```
xgb.train(params, xg_train, rounds, watchlist,
                        early_stopping_rounds=100, evals_result=evals)
```

However, is there an api for the scikitlearn facing one? I'm using the scikit learn one for cross validation testing and pipelining so it'd be very hard for me to change out of it.

If not, I can try contribute after some investigation.

Thanks!
